File: 태성/0603/로또의최고순위와최저순위.js
function solution(lottos, win_nums) {
    var answer = [];
    const dict = {
        0:6,
        1:6,
        2:5,
        3:4,
        4:3,
        5:2,
        6:1,
    }
    
    var max_cnt = 0
    for (let i=0; i<6; i++) {
        if (lottos[i] === 0) {
            max_cnt += 1
            
        } else{
            for(let j=0; j<6; j++) {
                if (lottos[i] === win_nums[j]) {
                    max_cnt += 1
                }
            }
        }
        
    }
    answer.push(dict[max_cnt])
    
    var min_cnt = 0
    for (let i=0; i<6; i++) {
        if (lottos[i] === 0) {
            
        } else{
            for(let j=0; j<6; j++) {
                if (lottos[i] === win_nums[j]) {
                    min_cnt += 1
                }
            }
        }
        
    }
    answer.push(dict[min_cnt])
    
    
    return answer;
}

File: 태성/0603/로또의최고순위와최저순위.py
def solution(lottos, win_nums):
    answer = []
    dic = {
        0:6,
        2:5,
        3:4,
        4:3,
        5:2,
        6:1,
    }
    max_cnt = 0
    for l in lottos:
        if l == 0:
            max_cnt += 1
        else:
            for w in win_nums:
                if l == w:
                    max_cnt += 1

    if max_cnt == 0 or max_cnt == 1:
        max_cnt = 0
    answer.append(dic[max_cnt])
    
    min_cnt = 0
    for l in lottos:
        for w in win_nums:
            if l == w:
                min_cnt += 1

    if min_cnt == 0 or min_cnt == 1:
        min_cnt = 0

    answer.append(dic[min_cnt])
    return answer